golang/pkg/dagent/update/update.go

```diff
@@ -0,0 +1,183 @@
+package update
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/dyrector-io/dyrectorio/golang/internal/helper/docker"
+	"github.com/dyrector-io/dyrectorio/golang/internal/helper/image"
+	"github.com/dyrector-io/dyrectorio/golang/internal/util"
+	"github.com/dyrector-io/dyrectorio/golang/pkg/dagent/utils"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+
+	containerbuilder "github.com/dyrector-io/dyrectorio/golang/pkg/builder/container"
+)
+
+var _selfUpdateDeadline *int64
+
+func getContainerName(ctx context.Context, base string) (string, error) {
+	name := base
+	count := 0
+
+	for {
+		container, err := docker.GetContainerByName(ctx, nil, name)
+		if err != nil {
+			return base, err
+		}
+		if len(container.Names) == 0 {
+			return name, nil
+		}
+
+		count++
+		name = base + fmt.Sprint(count)
+	}
+}
+
+func findImageAndPull(ctx context.Context, imageName string) (string, error) {
```

Comment:
I believe theres an imagepull func, if that's not adequate, please make it more general.

Reply:
`PullImage` force pulls the image that's why I decided to write this helper func.

Reply:
Pull image could be the same as this here, with an option to force pull. I like the implementation, this could all be alongside of the other docker util functions. However, that could be in an other PR as well, with a title like: `moving the rest of docker
 util functions`.